File: Java/ISL29003.java
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// ISL29003
// This code is designed to work with the ISL29003_I2CS I2C Mini Module available from ControlEverything.com.
// https://www.controleverything.com/content/Light?sku=ISL29003_I2CS#tabs-0-product_tabset-2

import com.pi4j.io.i2c.I2CBus;
import com.pi4j.io.i2c.I2CDevice;
import com.pi4j.io.i2c.I2CFactory;
import java.io.IOException;

public class ISL29003
{
	public static void main(String args[]) throws Exception
	{
		// Create I2C bus
		I2CBus bus = I2CFactory.getInstance(I2CBus.BUS_1);
		// Get I2C device, ISL29003 I2C address is 0x44(68)
		I2CDevice device = bus.getDevice(0x44);

		// Select command register
		// Normal operation, 16-bit resolution
		device.write(0x00, (byte)0x80);
		// Select control register
		// Gain = 64000 lux
		device.write(0x01, (byte)0x0C);
		Thread.sleep(500);

		// Read 2 bytes of data from address 0x04(4)
		// luminance lsb, luminance msb
		byte[] data = new byte[2];
		device.read(0x04, data, 0, 2);

		// Convert the data
		int luminance = ((data[1] & 0xFF) * 256 + (data[0] & 0xFF));
		
		// Output data to screen
		System.out.printf("Ambient Light Luminance : %d lux %n", luminance);
	}
}

File: Python/ISL29003.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# ISL29003
# This code is designed to work with the ISL29003_I2CS I2C Mini Module available from ControlEverything.com.
# https://www.controleverything.com/content/Light?sku=ISL29003_I2CS#tabs-0-product_tabset-2

import smbus
import time

# Get I2C bus
bus = smbus.SMBus(1)

# ISL29003 address, 0x44(68)
# Select command register, 0x00(00)
#		0x80(128)	Normal operation, 16-bit resolution
bus.write_byte_data(0x44, 0x00, 0x80)
# ISL29003 address, 0x44(68)
# Select control register, 0x01(01)
#		0x0C(12)	Gain = 64000 lux
bus.write_byte_data(0x44, 0x01, 0x0C)

time.sleep(0.5)

# ISL29003 address, 0x44(68)
# Read data back from 0x04(04), 2 bytes
# luminance LSB, luminance MSB
data = bus.read_i2c_block_data(0x44, 0x04, 2)

# Convert the data
luminance = data[1] * 256 + data[0]

# Output data to screen
print "Ambient Light luminance : %d lux" %luminance
